pkg/transfer/local/pull.go

```diff
@@ -241,3 +248,20 @@ func fetchHandler(ingester content.Ingester, fetcher remotes.Fetcher, pt *Progre
 		}
 	}
 }
+
+// getSupportedPlatform returns a matched platform comparing input UnpackConfiguration to the supported platform/snapshotter combinations
+// If input platform didn't specify snapshotter, default will be used if there is a match on platform.
+func getSupportedPlatform(uc image.UnpackConfiguration, supportedPlatforms []unpack.Platform) (bool, unpack.Platform) {
+	var u unpack.Platform
+	for _, sp := range supportedPlatforms {
+		//If both platform and snapshotter match, return the supportPlatform
+		//If platform matched and SnapshotterKey is empty, we assume client didn't pass SnapshotterKey
+		//use default Snapshotter
+		if sp.Platform.Match(uc.Platform) && sp.SnapshotterKey == uc.Snapshotter {
+			return true, sp
+		} else if sp.Platform.Match(uc.Platform) && sp.SnapshotterKey == "" {
```

Comment:
`uc.Snapshotter == ""` for the second condition as this would indicate the client has no snapshotter preference, only platform preference.

Reply:
Could also combine the first check to avoid calling `Match` twice.

Reply:
Good catch, updated in the new commit.